### PR TITLE
Python cleanup tasks 

### PR DESCRIPTION
**Pull request by samscott89**
Significant:
- Relaxed python requirements `cffi >= 1.0` and `django >= 2.2`
- Removed unused/undocumented extras - Http, Jwt, Pathmapper (but kept around for some tests)
- Excluded tests from the python packages

Cosmetic:
- Added linting using flake8 and type checking to `oso/polar` with mypy
- Fixed all the warnings

### CODE REVIEW

**Comment by saolsen** [comment]
Can you run a release job to test the cffi change on all platforms?

**Comment by saolsen** [comment]
Line 19 and 529 also have to be sam/python-cleanup for that to work.

File: languages/python/django-oso/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-oso~=0.8.2
-django~=3.0
+oso~=0.8
+django>=2.2
```
Comment (saolsen) [inline languages/python/django-oso/requirements.txt:2]: hmm, I really need to finish setting up the other repo tests stuff.

**Review by saolsen — APPROVED** [review]

File: languages/python/oso/requirements.txt [hunk 2]
```diff
@@ -1,2 +1 @@
-cffi==1.14.0
-dataclasses ; python_version < '3.7'
+cffi~=1.14
```
Comment (gj) [inline languages/python/oso/requirements.txt:1]: Is this as relaxed as we can make it?

**Pull request merged**